gcp-connector-util/main_windows.go

```diff
@@ -9,34 +9,189 @@
 package main
 
 import (
+	"fmt"
 	"log"
 	"os"
+	"path/filepath"
 
 	"github.com/codegangsta/cli"
 	"github.com/google/cups-connector/lib"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/eventlog"
+	"golang.org/x/sys/windows/svc/mgr"
 )
 
 var windowsCommands = []cli.Command{
 	cli.Command{
 		Name:      "init",
 		ShortName: "i",
-		Usage:     "Creates a config file",
+		Usage:     "Create a config file",
 		Action:    initConfigFile,
 		Flags:     commonInitFlags,
 	},
+	cli.Command{
+		Name:   "install-event-log",
+		Usage:  "Install registry entries for the event log",
+		Action: installEventLog,
+	},
+	cli.Command{
+		Name:   "remove-event-log",
+		Usage:  "Remove registry entries for the event log",
+		Action: removeEventLog,
+	},
+	cli.Command{
+		Name:   "create-service",
+		Usage:  "Create a service in the local service control manager",
+		Action: createService,
+	},
+	cli.Command{
+		Name:   "delete-service",
+		Usage:  "Delete an existing service in the local service control manager",
+		Action: deleteService,
+	},
+	cli.Command{
+		Name:   "start-service",
+		Usage:  "Start the service in the local service control manager",
+		Action: startService,
+	},
+	cli.Command{
+		Name:   "stop-service",
+		Usage:  "Stop the service in the local service control manager",
+		Action: stopService,
+	},
 }
 
 func updateConfig(config *lib.Config, configMap map[string]interface{}) bool {
 	return commonUpdateConfig(config, configMap)
 }
 
+func installEventLog(c *cli.Context) {
+	err := eventlog.InstallAsEventCreate(lib.ConnectorName, eventlog.Error|eventlog.Warning|eventlog.Info)
+	if err != nil {
+		fmt.Printf("Failed to install event log registry entries: %s\n", err)
+		os.Exit(1)
+	}
+	fmt.Println("Event log registry entries installed successfully")
+}
+
+func removeEventLog(c *cli.Context) {
+	err := eventlog.Remove(lib.ConnectorName)
+	if err != nil {
+		fmt.Printf("Failed to remove event log registry entries: %s\n", err)
+		os.Exit(1)
+	}
+	fmt.Println("Event log registry entries removed successfully")
+}
+
+func createService(c *cli.Context) {
+	exePath, err := filepath.Abs("gcp-windows-connector.exe")
+	if err != nil {
+		fmt.Printf("Failed to find the connector executable: %s\n", err)
+		os.Exit(1)
+	}
+
+	m, err := mgr.Connect()
+	if err != nil {
+		fmt.Printf("Failed to connect to service control manager: %s\n", err)
+		os.Exit(1)
+	}
+	defer m.Disconnect()
+
+	config := mgr.Config{
+		DisplayName:  lib.ConnectorName,
```

Comment:
No way to choose currently. If someone asks for it, then I'll add it. The default is SYSTEM.